### PR TITLE
removed non-empty results returned with non-nill errors

### PR DESCRIPTION
**Pull request by ngraham20**

### CODE REVIEW

**Comment by ngraham20** [comment]
Manual testing of MulticlusterHub Operator with this PR image doesn't reveal any issues. Overriding the sonarCloud due to all changes being related to API error blocks

File: api/v1/multiclusterhub_webhook_test.go [hunk 1]
```diff
@@ -116,13 +116,14 @@ var _ = Describe("Multiclusterhub webhook", func() {
...
-		})
+		// TODO: Somehow, `Expect(k8sClient.Delete(ctx, mch)).To(BeNil(), "MCH delete was blocked unexpectedly")` causes a strange error
+		// It("Should delete multiclusterhub", func() {
+		// 	By("deleting", func() {
+		// 		mch := &MultiClusterHub{}
+		// 		Expect(k8sClient.Get(ctx,
+		// 			types.NamespacedName{Name: multiClusterHubName, Namespace: "default"}, mch)).To(Succeed())
+		// 		Expect(k8sClient.Delete(ctx, mch)).To(BeNil(), "MCH delete was blocked unexpectedly")
+		// 	})
+		// })
```
Comment (dislbenn) [inline api/v1/multiclusterhub_webhook_test.go:127]: I'm not sure if we want to disable our test case for webhook deletion.

Comment (ngraham20) [reply api/v1/multiclusterhub_webhook_test.go:127]: I'll bring this up in scrum 👍 thanks for reminding me about this change!

**Pull request merged**